internal/services/compute/managed_disk_resource.go

```diff
@@ -980,7 +980,11 @@ func resourceManagedDiskDelete(d *pluginsdk.ResourceData, meta interface{}) erro
 	return nil
 }
 
-func shutDownOnResize(oldSizeGB, newSizeGB int) bool {
+func shutDownOnResize(disk compute.Disk, oldSizeGB, newSizeGB int) bool {
+	// OS disks can't be expanded without downtime.
+	if disk.OsType != "" {
```

Comment:
@tombuildsstuff - let me know if you have suggestions for a better way of detecting data disks.